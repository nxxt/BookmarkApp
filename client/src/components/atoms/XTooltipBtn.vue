<template>
  <v-tooltip bottom>
    <!-- btn -->
    <template v-slot:activator="{ on }">
      <div v-on="on">
        <v-btn
          class="x-tooltip-btn"
          :class="classes"
          v-bind="$attrs"
          v-on="listeners"
        >
          <slot></slot>
        </v-btn>
      </div>
    </template>

    <!-- tooltip content -->
    <span>{{ tooltip }}</span>
  </v-tooltip>
</template>

<script lang="ts">
import { PropType } from 'vue'
import { Component, Prop, Vue } from 'vue-property-decorator'

@Component({
  inheritAttrs: false
})
export default class XTooltipBtn extends Vue {
  @Prop({ type: [String, Number], default: 'tooltip' })
  tooltip!: string | number

  @Prop({ type: Array as PropType<String[]>, default: () => [] })
  classes!: string[]

  get listeners() {
    const vm = this
    return {
      ...vm.$listeners,
      click: (event: Event) => vm.$emit('click', event)
    }
  }
}
</script>
