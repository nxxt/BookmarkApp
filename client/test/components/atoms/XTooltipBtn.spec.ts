import Vuetify from 'vuetify'
import { mount, createLocalVue } from '@vue/test-utils'
import XTooltipBtn from '@/components/atoms/XTooltipBtn.vue'

const localVue = createLocalVue()

describe('XTooltipBtn.vue', () => {
  let vuetify: typeof Vuetify
  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const propsData = {
    tooltip: 'test',
    classes: ['test']
  }

  const factory = (options?: any) => {
    return mount(XTooltipBtn, {
      localVue,
      vuetify,
      ...options
    })
  }

  // TEST
  it('props', () => {
    const wrapper = factory({ propsData })
    expect(wrapper.props()).toStrictEqual(propsData)
  })

  describe('event', () => {
    const wrapper = factory()

    const event = jest.fn()
    wrapper.vm.$on('click', event)
    expect(event).toHaveBeenCalledTimes(0)

    wrapper.find('.v-btn').trigger('click')

    expect(event).toHaveBeenCalledTimes(1)
  })

  describe('the entire template', () => {
    it('slot', () => {
      const wrapper = factory({
        slots: {
          default: '<div data-test="slotContent">slot content</div>'
        }
      })

      const slotContent = wrapper.find('[data-test="slotContent"]')
      expect(slotContent.exists()).toBe(true)
      expect(slotContent.text()).toBe('slot content')
    })

    it('snapshot', () => {
      const wrapper = factory({ propsData })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
