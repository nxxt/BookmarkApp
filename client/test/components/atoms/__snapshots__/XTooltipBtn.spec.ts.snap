// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XTooltipBtn.vue the entire template snapshot 1`] = `"<span class=\\"v-tooltip v-tooltip--bottom\\"><!----><div><button type=\\"button\\" class=\\"x-tooltip-btn v-btn v-btn--contained theme--light v-size--default test\\"><span class=\\"v-btn__content\\"></span></button></div></span>"`;
